### PR TITLE
feat: Add client-side event truncation (100KB max)

### PR DESCRIPTION
**Pull request by naji247**
## Summary

- Adds a three-layer event truncation system (`src/modules/truncation.ts`) that guarantees no event exceeds 100KB before being sent to the MCPCat API
- **Layer 1:** Field-level limits (userIntent 2K, resourceName 256, metadata 256, error.message 2K, stack frames 50, response text 32KB)
- **Layer 2:** Recursive object normalization (depth 10, breadth 100, circular ref detection, non-serializable value coercion)
- **Layer 3:** Size-targeted fallback with progressive depth reduction and last-resort string truncation
- Integrates into the event queue pipeline after `sanitizeEvent()` — 2 lines added to `eventQueue.ts`
- 48 new tests covering all truncation behaviors, edge cases, non-mutation guarantees, and the 100KB size constraint

> **Depends on:** #25 (content sanitization) — this branch is based on `feat/content-sanitization`

## Test plan

- [x] All 48 truncation tests pass
- [x] Full test suite passes (369/369 + 1 pre-existing skip)
- [x] Lint clean
- [x] Typecheck clean
- [ ] Verify events are truncated in staging with oversized tool responses

### CODE REVIEW

File: src/modules/truncation.ts [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+  return textEncoder.encode(JSON.stringify(value)).length;
+}
+
+/**
+ * Finds and truncates the largest string values in an object to fit within a byte budget.
+ * Last-resort mechanism when depth reduction alone isn't enough.
+ * Iterates until the result fits or no further reduction is possible.
+ */
+function truncateLargestFields(obj: any, maxBytes: number): any {
+  let result = JSON.parse(JSON.stringify(obj)); // deep clone
```
Comment (kashishhora) [inline src/modules/truncation.ts]: Doing a deep clone here actually turns some typed objects like `Date` into `string`. Added a unit test to catch this.

**Review by kashishhora — CHANGES_REQUESTED** [review]

**Review by kashishhora — APPROVED** [review]
